File: modules/@angular/common/src/forms-deprecated/directives/normalize_validator.dart
library angular2.core.forms.normalize_validators;

import 'package:angular2/src/common/forms/directives/validators.dart' show Validator;

Function normalizeValidator(dynamic validator){
  if (validator is Validator) {
    return (c) => validator.validate(c);
  } else {
    return validator;
  }
}


Function normalizeAsyncValidator(dynamic validator){
  if (validator is Validator) {
    return (c) => validator.validate(c);
  } else {
    return validator;
  }
}

File: modules/@angular/forms/src/directives/normalize_validator.ts
import {AbstractControl} from '../model';

import {AsyncValidatorFn, Validator, ValidatorFn} from './validators';

export function normalizeValidator(validator: ValidatorFn | Validator): ValidatorFn {
  if ((<Validator>validator).validate !== undefined) {
    return (c: AbstractControl) => (<Validator>validator).validate(c);
  } else {
    return <ValidatorFn>validator;
  }
}

export function normalizeAsyncValidator(validator: AsyncValidatorFn | Validator): AsyncValidatorFn {
  if ((<Validator>validator).validate !== undefined) {
    return (c: AbstractControl) => Promise.resolve((<Validator>validator).validate(c));
  } else {
    return <AsyncValidatorFn>validator;
  }
}
